### PR TITLE
Support for MariaDB

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Support for MariaDB
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Sponsor company   | @PrestaShopCorp

## How to test?

* PR : #36295
* `pr_test_one.yml`
  * `mysql` : :green_circle:  https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514096143
  * `mariadb` : :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514094464
* `pr_test_single_campaign.yml` : `functional:API` for the test
  * `mysql`  :green_circle:  https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514197006
  * `mariadb` : :green_circle:  https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514201102
* `pr_test_single_campaign.yml` : `sanity`
  * `mysql`  : :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514205159
  * `mariadb` : :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514209613
* `pr_test.yml`
  * `mysql`  : :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514215013
  * `mariadb` :  :green_circle: https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/9514219305

### CODE REVIEW

File: .github/workflows/build-shop.yml [hunk 1]
```diff
@@ -80,13 +86,13 @@ jobs:
...
-          # Pull mysql image
-          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml pull -q mysql >& /dev/null &
+          # Pull ${{ inputs.db_type }} image
+          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml pull -q ${{ inputs.db_type }} >& /dev/null &
       - name: Build PrestaShop image in background
         working-directory: ${{ env.PS_DIR }}
         run: |
           # Build prestashop image in background
-          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml build >& /dev/null &
+          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml build prestashop-git >& /dev/null &
```
Comment (jolelievre) [inline .github/workflows/build-shop.yml:94]: I believe the dependency you setup in the docker file depends on the env variable DB_SERVICE/DB_SERVER I guess you could use the build without specifying `prestashop-git` if you set this env variable no?

File: .github/workflows/build-shop.yml [hunk 2]
```diff
@@ -205,7 +211,8 @@ jobs:
             "*/tests/Resources/**/*" \
             "*/tests/Unit/**/*" \
             "*/.git/**/*"
-          docker exec my_prestashop-mysql-1 /usr/bin/mysqldump -u ${{ env.DB_USER }} -p${{ env.DB_PASSWD }} ${{ env.DB_NAME }} > /tmp/shop-artifacts/db_dump.sql
+          docker exec my_prestashop-${{ inputs.db_type }}-1 ${{ (inputs.db_type == 'mysql') && '/usr/bin/mysqldump'|| 'mariadb-dump' }} --version
```
Comment (jolelievre) [inline .github/workflows/build-shop.yml:213]: Is this line for debugging?

File: .github/workflows/pr_security_test_one.yml [hunk 3]
```diff
@@ -49,6 +49,14 @@ on:
           - '14.21.3'
           - '16.20.1'
         default: '16.20.1'
+      DB_TYPE:
```
Comment (jolelievre) [inline .github/workflows/pr_security_test_one.yml:52]: Couldn't we keep the same variable everywhere for consistency? It's a bit hard to follow between DB_TYPE, DATABASE, DB_SERVICE, DB_SERVER 🤪

File: .github/workflows/test-sanity.yml [hunk 4]
```diff
@@ -78,16 +82,16 @@ jobs:
...
-          # Pull mysql image
-          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml pull -q mysql >& /dev/null &
+          # Pull ${{ inputs.db_type }} image
+          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml pull -q ${{ inputs.db_type }}  >& /dev/null &
       - name: Build PrestaShop image in background
         working-directory: ${{ env.PS_DIR }}
         run: |
           # Build prestashop image in background
-          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml build >& /dev/null &
+          USER_ID=$(id -u) GROUP_ID=$(id -g) nohup docker compose -f docker-compose.yml build prestashop-git >& /dev/null &
```
Comment (jolelievre) [inline .github/workflows/test-sanity.yml:94]: Same question here, isn't there an env variable that is supposed to manage this correctly? Or is it that the default behaviour is to build all the defined services?
Maybe we can prevent this by adding `required: false` on both DB services

**Review by matthieu-rolland — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
